tgui/packages/tgui/interfaces/SpawnPanel/CreateObject.tsx

```diff
@@ -0,0 +1,453 @@
+import { storage } from 'common/storage';
+import { useEffect, useState } from 'react';
+import {
+  Box,
+  Button,
+  DmIcon,
+  Section,
+  Stack,
+  VirtualList,
+} from 'tgui-core/components';
+import { useFuzzySearch } from 'tgui-core/fuzzysearch';
+
+import { useBackend } from '../../backend';
+import { SearchBar } from '../common/SearchBar';
+import { listNames, listTypes } from './constants';
+import { CreateObjectSettings } from './CreateObjectSettings';
+import { AtomData, CreateObjectProps } from './types';
+
+interface SpawnPanelData {
+  icon: string;
+  iconState: string;
+  selected_object?: string;
+  copied_type?: string;
+  preferences?: {
+    hide_icons: boolean;
+    hide_mappings: boolean;
+    sort_by: string;
+    search_text: string;
+    search_by: string;
+  };
+}
+
+interface SpawnPreferences {
+  hide_icons: boolean;
+  hide_mappings: boolean;
+  sort_by: string;
+  search_text: string;
+  search_by: string;
+  object_list?: string;
+}
+
+interface StateSetterConfig<T extends unknown> {
+  value: T;
+  storageKey: string;
+  setter: (value: T) => void;
+}
+
+const setStateAndStorage = async <T extends unknown>({
+  value,
+  storageKey,
+  setter,
+}: StateSetterConfig<T>) => {
+  setter(value);
+  await storage.set(storageKey, value);
+};
+
+interface CurrentList {
+  [key: string]: AtomData;
+}
+
+export function CreateObject(props: CreateObjectProps) {
+  const { act, data } = useBackend<SpawnPanelData>();
+  const { objList, setAdvancedSettings, iconSettings } = props;
```

Comment:
if you create a default of `{}` for objList, you wouldn't need to do some of the later acrobatics like in L78